scripts/send-bundles-sizes-differences.js

```diff
@@ -0,0 +1,181 @@
+  }
+  const comments = await response.json()
+  const targetComment = comments.filter((comment) => comment.body.startsWith('## Bundles Sizes Evolution'))[0]
+  if (targetComment !== undefined) {
+    return targetComment.id
+  }
+}
+
+function loadQuery(commitSha) {
+  return [
+    {
+      name: 'bundles_sizes_rum',
+      queries: `avg:bundle_sizes.rum{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_logs',
+      queries: `avg:bundle_sizes.logs{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_rum_slim',
+      queries: `avg:bundle_sizes.rum_slim{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_worker',
+      queries: `avg:bundle_sizes.worker{commit:${commitSha}}`,
+    },
+  ]
+}
+
+async function queryWithRetry(budget, retries = 4) {
+  const now = Math.floor(Date.now() / 1000)
+  const date = now - 30 * 24 * 60 * 60
+  for (let i = 0; i < retries; i++) {
+    const response = await fetch(
+      `https://api.datadoghq.com/api/v1/query?query=${budget.queries}&from=${date}&to=${now}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          'DD-API-KEY': getOrg2ApiKey(),
+          'DD-APPLICATION-KEY': getOrg2AppKey(),
+        },
+      }
+    )
+    if (!response.ok) {
+      throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+    }
+    const data = await response.json()
+    if (data.series && data.series.length > 0 && data.series[0].pointlist && data.series[0].pointlist.length > 0) {
+      return {
+        name: budget.name,
+        size: data.series[0].pointlist[0][1],
+      }
+    }
+    await timeout(5000)
+  }
+  return {
+    name: budget.name,
+    size: null,
+  }
+}
+
+function query(results) {
+  return Promise.all(results.map((budget) => queryWithRetry(budget)))
+}
+
+function compare(resultsBaseQuery, resultsLocalQuery) {
+  return resultsBaseQuery.map((baseResult, index) => {
+    const localResult = resultsLocalQuery[index]
+    let percentageChange = null
+
+    if (baseResult.size && localResult.size) {
+      percentageChange = (((localResult.size - baseResult.size) / baseResult.size) * 100).toFixed(2)
+    }
+    return {
+      name: baseResult.name,
+      percentageChange,
+    }
+  })
+}
+
+function formatName(bundleName) {
+  return bundleName
+    .split('_')
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ')
+}
+
+runMain(async () => {
+  const lastCommonCommit = getLastCommonCommit(baseBranch)
+  const latestLocalCommit = process.env.CI_COMMIT_SHORT_SHA
+  const prs = await getPRs(localBranch)
+  if (prs.length === 0) {
+    console.log('No pull requests found')
+    return
+  }
+  const resultsBaseQuery = await query(loadQuery(lastCommonCommit))
+  const resultsLocalQuery = await query(loadQuery(latestLocalCommit))
+  const difference = compare(resultsBaseQuery, resultsLocalQuery)
+  const commentId = await getCommentId(prs[0].number)
```

Comment:
It could worth double checking but if we don't expect to retrieve multiple PRs, what about:

- throwing an error if it happens rather than silently updating only one of them
- work with either a single PR or no PR inside the main to keep things simple

Reply:
I made 2 case (equal to 0, or greater than 1) to identify the error rather than just proceed if the prs.length is equal to 1

Reply:
```js
const prs = await getPRs(LOCAL_BRANCH)
if (prs.length === 0) {
    console.log('No pull requests found for the branch')
    process.exit(0)
}
if (prs.length > 1) {
    throw new Error('More than one PR found for the branch')
}
```

From the main perspective, it could be clearer to always expect one PR and we could throw the error for multiple PRs from the function that fetch the PR:
```js
const pr = await fetchPr(LOCAL_BRANCH)
if (!pr) {
    console.log('No pull requests found for the branch')
    process.exit(0)
}
```
It would avoid to clutter the main function with the handling of this unexpected error.